### PR TITLE
use github recognized CODEOWNERS file

### PR DESCRIPTION
**Pull request by timotheecour**
Note: I'm aware there are pros and cons with this PR.

It turns out github understands CODEOWNERS file, using a syntax similar to gitignore:  https://help.github.com/articles/about-codeowners/
this PR converts doc/codeowners.rst to something github will understand.

## Benefits
Benefits as described in that link, eg PR's are automatically sent for review (ie marking people as reviewers) to the "most specific" people matching a pattern.

## Drawback
Drawback is it only works with gitignore patterns, hence the section marked `informal entries not easily mapped to CODEOWNERS` : the mapping is specified as comment instead of github codeowner syntax
(not really a drawback since existing doc/codeowners.rst is pure documentation anyways, but I guess the drawback is we lose the nice formatting as a table and instead have this file)

### CODE REVIEW

File: .github/CODEOWNERS [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+#incremental:                    @araq
+#sizeof computations:            @krux02
+#Exception handling:             @cooldome @araq
+
+## Standard library
+
+# io: @dom96
+# garbage collector: @araq
+# Go garbage collector: @stefantalpalaru
+# coroutines:  @Rokas @Kupstys
```
Comment (Araq) [inline .github/CODEOWNERS]: His github name is ``@rokups``

**Comment by Araq** [comment]
I still don't understand how github uses this information. How does it work?

Also, I still prefer a "Code owner" comment in the source files, feels much easier to maintain.

File: .github/CODEOWNERS (hunk 1 above)
Comment (timotheecour) [reply .github/CODEOWNERS]: fixed; also removed @Kupstys which I'm not seeing in https://github.com/Kupstys

**Comment by timotheecour** [comment]
> I still don't understand how github uses this information. How does it work?

quote from https://help.github.com/articles/about-codeowners/ : 
> Code owners are automatically requested for review when someone opens a pull request that modifies code that they own

so it makes the process of assigning for review automatic instead of what we have now; should we try it for a week and see how it goes? we can always revert this PR if things don't work out

**Comment by Araq** [comment]
> so it makes the process of assigning for review automatic instead of what we have now; should we try it for a week and see how it goes?

And how does github know the "VM" subsystem consists of "vm.nim, vmgen.nim", etc? Well you said it yourself, "informal entries not easily mapped to CODEOWNERS"

**Comment by timotheecour** [comment]
> And how does github know the "VM" subsystem consists of "vm.nim, vmgen.nim", etc? Well you said it yourself, "informal entries not easily mapped to CODEOWNERS"

subsequent PR's can add formal entries wherever possible, I didn't want to "guess" what each subsystem was in this PR; eg:
```
vm.nim @foo
vmgen.nim @foo
# may be incomplete...
```

other cases it may be harder to do, I reckon, but it's not really any worse than what we have now where we need to manually assign; at least here all formal entries would be automatic.

I also reckon that github's format is not as good as it could've been eg this would've been better:
```nim
# regex : comma-separated @names
foo/(bar|baz.nim)|t?baz.nim : @foo1, @foo2
```
another option would be to automatically generate a CODEOWNERS in github format from the above format, and update on each commit via a git hook; what do you think.

**Comment by Araq** [comment]
If it doesn't work anyway, we might as well choose my solution and add comments to the .nim files, which is at least maintainable.

**Comment by timotheecour** [comment]
> If it doesn't work anyway, we might as well choose my solution and add comments to the .nim files, which is at least maintainable.

well that's what I'm seeing here in this recent commit: f84bf45f17d2ec441ea455bd0387d11e4124654f by @krux02 
which adds code owner info to ./compiler/sizealignoffsetimpl.nim
```
+## code owner: Arne Döring
+## e-mail: arne.doering@gmx.net
```
and I don't think that scales well. At very least that commit should've just been:
```
+## codeowner: @krux02 
```
the rest being irrelevant/not DRY (you don't want to duplicate name and email ; eg in case needs to change etc)

All in 1 place is better, and works better with tooling.
the github approach (which I'm doing in this PR) results in less maintenance since we can use globs on files/directories.

**Comment by krux02** [comment]
I disagree. The comment isn't supposed to work for tooling it is supposed to work for humans who read the file.

**Pull request closed without merging**